rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -257,6 +257,11 @@ void SequentialWriter::switch_to_next_storage()
   for (const auto & topic : topics_names_to_info_) {
     storage_->create_topic(topic.second.topic_metadata);
   }
+
+  if (use_cache_) {
```

Comment:
At first glance this looks incorrect since the callback needs to be changed. I see that the snapshot changes broke the cache mechanism and were apparently not tested with splitting.

Edit: After looking into previous PR I believe this is ok since the callback is now a member function and it is ensured that the callback is always good.

Reply:
In which case, I think it'd be best not to bounce the consumer thread on every split. I won't block on it though.

Reply:
@hidmic When we are doing bag split we are recreating `storage_` object and we need to prevent calling `staorge_->write(..)` from any other threads. 
There are two options stop/start `cache->consumer` thread or add extra lock inside consumer thread triggering outside i.e. from `SequentialWriter`. 
The last option is less preferable IMO, since it will over-complicate design and will be more prone for the deadlocks and race conditions.